GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -74,59 +72,43 @@ - (NSURL *)directoryPathURL {
  * @param directoryPathURL The path to the directory that needs to exist.
  */
 - (void)checkAndCreateDirectory:(NSURL *)directoryPathURL {
-  NSError *fileCoordinatorError = nil;
-  [self.fileCoordinator
-      coordinateWritingItemAtURL:directoryPathURL
-                         options:0
-                           error:&fileCoordinatorError
-                      byAccessor:^(NSURL *writingDirectoryURL) {
-                        NSError *error;
-                        if (![writingDirectoryURL checkResourceIsReachableAndReturnError:&error]) {
-                          NSError *error;
-                          [[NSFileManager defaultManager] createDirectoryAtURL:writingDirectoryURL
-                                                   withIntermediateDirectories:YES
-                                                                    attributes:nil
-                                                                         error:&error];
-                        }
-                      }];
+  NSError *error;
+  if (![directoryPathURL checkResourceIsReachableAndReturnError:&error]) {
+    NSError *error;
+    [[NSFileManager defaultManager] createDirectoryAtURL:directoryPathURL
+                             withIntermediateDirectories:YES
+                                              attributes:nil
+                                                   error:&error];
+  }
 }
 
 - (nullable NSDate *)heartbeatDateForTag:(NSString *)tag {
-  __block NSDictionary *heartbeatDictionary;
-  NSError *error;
-  [self.fileCoordinator coordinateReadingItemAtURL:self.fileURL
-                                           options:0
-                                             error:&error
-                                        byAccessor:^(NSURL *readingURL) {
-                                          heartbeatDictionary =
-                                              [self heartbeatDictionaryWithFileURL:readingURL];
-                                        }];
-  NSDate *heartbeatDate = heartbeatDictionary[tag];
-  if (![heartbeatDate isKindOfClass:[NSDate class]]) {
-    return nil;
+  @synchronized(self.class) {
+    NSDictionary *heartbeatDictionary = [self heartbeatDictionaryWithFileURL:self.fileURL];
+    NSDate *heartbeatDate = heartbeatDictionary[tag];
+    if (![heartbeatDate isKindOfClass:[NSDate class]]) {
```

Comment:
What else can `heartbeatDate` be except for an `NSDate`?

Reply:
The intention of this check is handle file format updates from newer versions gracefully. If the file format is unexpected we ignore its content (instead of e.g. returning invalid type or crashing).

Reply:
Thanks, perhaps that justifies a comment?